runtime/src/iree/hal/drivers/cuda/cuda_device.c

```diff
@@ -184,6 +184,20 @@ iree_status_t iree_hal_cuda_device_create(
   return status;
 }
 
+iree_status_t iree_hal_cuda_device_get_context(iree_hal_device_t* base_device,
+                                               CUcontext* out_context) {
+  IREE_ASSERT_ARGUMENT(out_context);
+
+  if (!iree_hal_resource_is(base_device, &iree_hal_cuda_device_vtable)) {
+    return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                            "HAL device is not a CUDA device");
+  }
+
+  iree_hal_cuda_device_t* device = iree_hal_cuda_device_cast(base_device);
```

Comment:
I originally added two functions: 1. is_cuda_device 2. get_cuda_context, but turned them into single function because it's harder to misuse and forget about the check, and get a segfault at run time.